File: packages/sdk/go/internal/model/memory.go
// Package model 的记忆(保存 / 召回 / 搜索 / Dream / 分类记忆)类型。
package model

// SaveMemoryRequest 是 POST /v1/memory 请求体(保存记忆)。
type SaveMemoryRequest struct {
	Content  string         `json:"content"`
	Type     string         `json:"type,omitempty"`
	Metadata map[string]any `json:"metadata,omitempty"`
}

// SaveMemoryResponse 是保存记忆响应。
type SaveMemoryResponse struct {
	MemoryID string `json:"memoryId"`
	Status   string `json:"status"`
}

// RecallMemoryItem 是召回记忆项。
type RecallMemoryItem struct {
	ID        string         `json:"id"`
	Content   string         `json:"content"`
	Type      string         `json:"type"`
	Score     float64        `json:"score"`
	CreatedAt string         `json:"createdAt"`
	Metadata  map[string]any `json:"metadata,omitempty"`
}

// RecallMemoryResponse 是 GET /v1/memory 响应体(召回记忆)。
type RecallMemoryResponse struct {
	Object string             `json:"object"`
	Data   []RecallMemoryItem `json:"data"`
}

// MemorySearchRequest 是 POST /v1/memory/search 请求体(语义搜索)。
type MemorySearchRequest struct {
	Query string `json:"query"`
	TopK  *int   `json:"topK,omitempty"`
	Type  string `json:"type,omitempty"`
}

// MemorySearchResponse 是 POST /v1/memory/search 响应体。
type MemorySearchResponse struct {
	Object string             `json:"object"`
	Data   []RecallMemoryItem `json:"data"`
}

// MemoryDreamRequest 是 POST /v1/memory/dream 请求体(Dream 梦境系统)。
type MemoryDreamRequest struct {
	Mode string `json:"mode,omitempty"`
}

// MemoryDreamResponse 是 POST /v1/memory/dream 响应体。
type MemoryDreamResponse struct {
	DreamID     string   `json:"dreamId"`
	Insights    []string `json:"insights"`
	NewMemories int      `json:"newMemories"`
}

// ForgetMemoryRequest 是 DELETE /v1/memory 请求体(遗忘记忆)。
type ForgetMemoryRequest struct {
	MemoryID string `json:"memoryId"`
}

// ForgetMemoryResponse 是遗忘记忆响应。
type ForgetMemoryResponse struct {
	MemoryID string `json:"memoryId"`
	Status   string `json:"status"`
}

// WorkingMemoryItem 是工作记忆项。
type WorkingMemoryItem struct {
	ID        string `json:"id"`
	Content   string `json:"content"`
	CreatedAt string `json:"createdAt"`
}

// WorkingMemoryResponse 是 GET /v1/memory/working 响应体。
type WorkingMemoryResponse struct {
	Items []WorkingMemoryItem `json:"items"`
}

// EpisodicMemoryItem 是情景记忆项。
type EpisodicMemoryItem struct {
	ID           string   `json:"id"`
	Summary      string   `json:"summary"`
	Timestamp    string   `json:"timestamp"`
	Participants []string `json:"participants"`
}

// EpisodicMemoryResponse 是 GET /v1/memory/episodic 响应体。
type EpisodicMemoryResponse struct {
	Episodes []EpisodicMemoryItem `json:"episodes"`
}

// ProceduralMemoryItem 是程序记忆项。
type ProceduralMemoryItem struct {
	ID          string   `json:"id"`
	Name        string   `json:"name"`
	Steps       []string `json:"steps"`
	SuccessRate float64  `json:"successRate"`
}

// ProceduralMemoryResponse 是 GET /v1/memory/procedural 响应体。
type ProceduralMemoryResponse struct {
	Procedures []ProceduralMemoryItem `json:"procedures"`
}

File: sdks/typescript/src/memory.ts
/**
 * 记忆模块 — 保存 / 召回 / 搜索 / Dream / 遗忘 / 分类记忆。
 *
 * 端点(8 个):
 * - POST   /v1/memory(保存记忆)
 * - GET    /v1/memory(召回记忆)
 * - POST   /v1/memory/search(语义搜索)
 * - POST   /v1/memory/dream(Dream 梦境系统)
 * - DELETE /v1/memory(遗忘记忆)
 * - GET    /v1/memory/working(工作记忆)
 * - GET    /v1/memory/episodic(情景记忆)
 * - GET    /v1/memory/procedural(程序记忆)
 */

import type { BaseClient } from './base.js'

// =============================================================================
// 内联类型定义
// =============================================================================

export interface V1SaveMemoryRequest {
  content: string
  type?: string
  metadata?: Record<string, unknown>
}

/** 保存记忆响应。 */
export interface V1SaveMemoryResponse {
  memoryId: string
  status: 'saved'
}

export interface V1RecallMemoryResponse {
  object: 'list'
  data: Array<{
    id: string
    content: string
    type: string
    score: number
    createdAt: string
    metadata?: Record<string, unknown>
  }>
}

export interface V1MemorySearchRequest {
  query: string
  topK?: number
  type?: string
}

/** 记忆搜索响应。 */
export interface V1MemorySearchResponse {
  object: 'list'
  data: Array<{
    id: string
    content: string
    type: string
    score: number
    createdAt: string
  }>
}

export interface V1MemoryDreamRequest {
  mode?: string
}

export interface V1MemoryDreamResponse {
  dreamId: string
  insights: string[]
  newMemories: number
}

export interface V1WorkingMemoryResponse {
  items: Array<{ id: string; content: string; createdAt: string }>
}

export interface V1EpisodicMemoryResponse {
  episodes: Array<{
    id: string
    summary: string
    timestamp: string
    participants: string[]
  }>
}

export interface V1ProceduralMemoryResponse {
  procedures: Array<{
    id: string
    name: string
    steps: string[]
    successRate: number
  }>
}

/** 遗忘记忆请求/响应。 */
export interface V1ForgetMemoryRequest {
  memoryId: string
}

export interface V1ForgetMemoryResponse {
  memoryId: string
  status: 'forgotten'
}

// =============================================================================
// Module 接口 + 工厂函数
// =============================================================================

export interface MemoryModule {
  /** POST /v1/memory(保存记忆)。 */
  save(req: V1SaveMemoryRequest): Promise<V1SaveMemoryResponse>
  /** GET /v1/memory(召回记忆)。 */
  recall(): Promise<V1RecallMemoryResponse>
  /** POST /v1/memory/search(语义搜索)。 */
  search(req: V1MemorySearchRequest): Promise<V1MemorySearchResponse>
  /** POST /v1/memory/dream(Dream 梦境系统)。 */
  dream(req?: V1MemoryDreamRequest): Promise<V1MemoryDreamResponse>
  /** DELETE /v1/memory(遗忘记忆)。 */
  forget(req: V1ForgetMemoryRequest): Promise<V1ForgetMemoryResponse>
  /** GET /v1/memory/working(工作记忆)。 */
  working(): Promise<V1WorkingMemoryResponse>
  /** GET /v1/memory/episodic(情景记忆)。 */
  episodic(): Promise<V1EpisodicMemoryResponse>
  /** GET /v1/memory/procedural(程序记忆)。 */
  procedural(): Promise<V1ProceduralMemoryResponse>
}

export function createMemoryModule(client: BaseClient): MemoryModule {
  return {
    save: (req) => client.request<V1SaveMemoryResponse>('POST', '/memory', req),
    recall: () => client.request<V1RecallMemoryResponse>('GET', '/memory'),
    search: (req) => client.request<V1MemorySearchResponse>('POST', '/memory/search', req),
    dream: (req) => client.request<V1MemoryDreamResponse>('POST', '/memory/dream', req),
    forget: (req) => client.request<V1ForgetMemoryResponse>('DELETE', '/memory', req),
    working: () => client.request<V1WorkingMemoryResponse>('GET', '/memory/working'),
    episodic: () => client.request<V1EpisodicMemoryResponse>('GET', '/memory/episodic'),
    procedural: () => client.request<V1ProceduralMemoryResponse>('GET', '/memory/procedural'),
  }
}